src/viam/sdk/common/proto_type.cpp

```diff
@@ -17,11 +17,14 @@ namespace sdk {
 using google::protobuf::Struct;
 using google::protobuf::Value;
 
+// NOLINTBEGIN(readability-redundant-declaration)
 Struct map_to_struct(AttributeMap dict);
 AttributeMap struct_to_map(Struct struct_);
+// NOLINTEND(readability-redundant-declaration)
 
 // float, string, struct, array, struct(map from string to any of the above)
 
+// NOLINTNEXTLINE(misc-no-recursion)
```

Comment:
I didn't choose to keep `misc-no-recursion` because recursion is bad, but rather because it is nice to warn the reader that this function is recursive.